libraries/exoplayer/src/main/java/androidx/media3/exoplayer/drm/DefaultDrmSession.java

```diff
@@ -644,6 +651,19 @@ public void handleMessage(Message msg) {
             break;
           case MSG_KEYS:
             response = callback.executeKeyRequest(uuid, (KeyRequest) requestTask.request);
+            if (currentKeyLoadInfo != null) {
+              String requestUrl = ((KeyRequest) requestTask.request).getLicenseServerUrl();
```

Comment:
This needs some work, the knowledge of URL's and such is really all private to the `MediaDrmCallback` so either we need to:

1. Ignore this and provide vacuous URL's (which, unless the `KeyRequest` has the data this is already true)
2. add methods to `MediaDrmInfo` to fill in the `LoadEventInfo`

Our use case does not care about the URL, simply the load times and retries.  So either way is good.

Reply:
@icbaker, thanks all great ideas...  How about we just let the `MediaDrmCallback` built the object, most of the info is already in the `HttpMediaDrmCallback`, the other implementations `MediaDrmCallback ` don't have to change as the default (null return) makes sense for them.

```java

  /**
   * Get the {@link LoadEventInfo} for the last executed request.
   * This can be null if no load is performed to execute the request
   *
   * @return the {@link LoadEventInfo} or null if no load was performed
   */
  @Nullable default LoadEventInfo getLastLoadEventInfo() { return null; }
  
 ````

With this, the `LoadEventInfo` is easy to produce so it is completely internally consistent (redirect included).

BTW, a few WV License Proxies we work with use delayed re-directs to manage load (Apache Traffic Server's "thundering herd" for example)

Reply:
See latest commit, adds this method.   With the `LoadEventInfo` produced in the object responsible for the actual load request it is completely consistant with what `Loader` would produce.